src/mscorlib/src/System/AppDomain.cs

```diff
@@ -1895,7 +1895,7 @@ public int ExecuteAssembly(String assemblyFile)
             RuntimeAssembly assembly = (RuntimeAssembly)Assembly.LoadFrom(assemblyFile, assemblySecurity);
     
             if (args == null)
-                args = new String[0];
+                args = Array.Empty<String>();
```

Comment:
AppDomains aren't exposed in .NET Core, and even if they were, the use of ExecuteAssembly is so rare relative to the costs involved that these empty string allocations shouldn't be at all noticeable.  We'd need to verify that nothing would be negatively affected by this kind of cached array usage, and while it's unlikely to harm anything, there's so little benefit that it's really not worth the time to do the investigation.  Let's revert all of the changes to this file.